.github/workflows/python-app.yml

```diff
@@ -15,10 +15,14 @@ jobs:
       uses: actions/setup-python@v2
       with:
         python-version: 3.8
+    - name: Run pre-commit
+      run: |
+        python3 -m pip install pre-commit
+        pre-commit run -A
     - name: Install PyYAML
       run: |
-         pip3 install -r requirements.txt
-         pip3 install coverage
+        pip3 install -r requirements.txt
+        pip3 install coverage
```

Comment:
@aswaterman In my view, and also thinks of a best practice and approach here, that reordering of the steps: Installing Dependencies and Running Pre-Commit -  to be interchanged.

What I meant to say that, installing dependencies before running pre-commit hooks, will ensure that any packages required by the hooks are available, preventing potential failures during the pre-commit phase.

I think the sequence to be like:
```
# First Installation then pre commit

name: Install PyYAML
      run: |
         pip3 install -r requirements.txt
         pip3 install coverage

# Now run pre commit
 name: Run pre-commit
      run: |
        python3 -m pip install pre-commit

```

Reply:
Yep that makes sense. I'll change it.

Reply:
Done